updater/aws_test.go

```diff
@@ -270,6 +292,197 @@ func TestFilterBottlerocketInstances(t *testing.T) {
+		ec2ID := "ec2-id-br" + strconv.Itoa(i)
+		instances = append(instances, aws.String(ec2ID))
+		descOut = append(descOut, &ecs.ContainerInstance{
+			Attributes:           []*ecs.Attribute{{Name: aws.String("bottlerocket.variant")}},
+			ContainerInstanceArn: aws.String(instanceARN),
+			Ec2InstanceId:        aws.String(ec2ID),
+		})
+	}
+
+	responses := []struct {
+		inputLen           int
+		ContainerInstances []*ecs.ContainerInstance
+		err                error
+	}{{
+		100,
+		descOut[:100],
+		errors.New("Failed to describe container instances"),
+	}, {
+		50,
+		descOut[100:],
+		errors.New("Failed to describe container instances"),
+	}}
+
+	callCount := 0
+	mockECS := MockECS{
+		DescribeContainerInstancesFn: func(input *ecs.DescribeContainerInstancesInput) (*ecs.DescribeContainerInstancesOutput, error) {
+			require.Less(t, callCount, len(responses))
+			resp := responses[callCount]
+			callCount++
+			assert.Equal(t, resp.inputLen, len(input.ContainerInstances))
+			return &ecs.DescribeContainerInstancesOutput{ContainerInstances: resp.ContainerInstances}, resp.err
+		},
+	}
+
+	u := updater{ecs: mockECS}
+	actual, err := u.filterBottlerocketInstances(instances)
+	assert.Empty(t, actual)
+	assert.Contains(t, err.Error(), "Failed to describe container instances")
+	// assert.EqualError(t, err, "failed to describe any container instances")
+}
+
+func TestPaginatedFilterBottlerocketInstancesSingleFailure(t *testing.T) {
+	descOut := make([]*ecs.ContainerInstance, 0)
+	instances := make([]*string, 0)
+	expected := make([]instance, 0)
+	for i := 0; i < 150; i++ {
+		instanceARN := "cont-inst-br" + strconv.Itoa(i)
+		ec2ID := "ec2-id-br" + strconv.Itoa(i)
+		instances = append(instances, aws.String(ec2ID))
+		descOut = append(descOut, &ecs.ContainerInstance{
+			Attributes:           []*ecs.Attribute{{Name: aws.String("bottlerocket.variant")}},
+			ContainerInstanceArn: aws.String(instanceARN),
+			Ec2InstanceId:        aws.String(ec2ID),
+		})
+		expected = append(expected, instance{
+			instanceID:          ec2ID,
+			containerInstanceID: instanceARN,
+		})
+	}
+
+	responses := []struct {
+		inputLen           int
+		ContainerInstances []*ecs.ContainerInstance
+		err                error
+	}{{
+		100,
+		nil,
+		errors.New("Failed to describe container instances"),
+	}, {
+		50,
+		descOut[100:],
+		nil,
+	}}
+
+	callCount := 0
+	mockECS := MockECS{
+		DescribeContainerInstancesFn: func(input *ecs.DescribeContainerInstancesInput) (*ecs.DescribeContainerInstancesOutput, error) {
+			require.Less(t, callCount, len(responses))
+			resp := responses[callCount]
+			callCount++
+			assert.Equal(t, resp.inputLen, len(input.ContainerInstances))
+			return &ecs.DescribeContainerInstancesOutput{ContainerInstances: resp.ContainerInstances}, resp.err
+		},
+	}
+
+	u := updater{ecs: mockECS}
+	actual, err := u.filterBottlerocketInstances(instances)
+	require.NoError(t, err)
+	assert.EqualValues(t, expected[100:], actual)
```

Comment:
// message here can be useful to understand the test.
```suggestion
	assert.EqualValues(t, expected[100:], actual, "should contain only last 50 instances")
```